lib/collection/src/collection_manager/optimizers/segment_optimizer.rs

```diff
@@ -193,14 +193,11 @@ pub trait SegmentOptimizer {
             sparse_vector_data
                 .iter_mut()
                 .for_each(|(vector_name, config)| {
-                    // Assign sparse index on disk
-                    if let Some(sparse_config) = &collection_params.sparse_vectors {
-                        if let Some(params) = sparse_config.get(vector_name) {
-                            if let Some(index) = params.index.as_ref() {
-                                config.index = Some(*index);
-                            }
-                        }
-                    }
+                    config
+                        .index
+                        .get_or_insert_with(Default::default)
+                        .on_disk
+                        .replace(true);
```

Comment:
Here we've reached the on-disk threshold. Instead of changing the global collection configuration to achieve this, we need to change the configuration for this segment being optimized.